### DISCUSSION TITLE
How to discover a reactivated device?

### DISCUSSION THREAD

**Discussion by nicknol** [andrew-codechimp/HA-Battery-Notes#1426] [Q&A] [answered]
I have a set of sensors that I only use in not-winter time (outdoor sensors). I deactivated them last autumn, in winter I detected and started to use battery notes.
I submitted the device to Andrew's library, I double checked and yes it's there.
Today I reactivated the devices but they are not discovered by battery notes.
=> how could I trigger the discovery? Is there a service call?

**Comment by andrew-codechimp** [comment, marked as answer]
Battery notes will perform discovery on a HA restart and also every 24 hours from a restart, mainly to detect if a new battery type is in the library but this will also work for new devices.

If they're still not showing up, try a manual add, you should be able to choose the device in the first screen and it will pick up the battery type from the library in the second.  If not showing in the list on the first screen then the device isn't supported, if battery type is not automatically filled in the second screen then there's something not quite right on the make/model in the library.

**Reply by nicknol** [reply]
I have to admit that I was a bit impatient and restarted HA 30 hours after I reactivated the devices. The devices showed up as new integrations and the information from the library was proposed.

Before I tried to add them manually but the information from the library wasn't proposed.

Anyways, they work for me now. ☺

I will be more patient when I have a similar case.

and: big thanks to you @andrew-codechimp 🥇

**Reply by hrabbach** [reply]
Hi, I know this is an old discussion, but I couldn't find a newer one and this one and the answer provided is the closest to my question/problem. I have recently (as in: over the last couple of days) added a number of new Zigbee devices. They are all supported, and I have other ones of the same type (Aqara door/window sensors and mini-buttons) already set up with Battery Notes that were automatically detected when I first set up Battery Notes. However, the newly added ones are not being detected (also not after several days, and also not after a HA restart). If I try to add them manually, contrary to what the above answer suggests, the battery type is not being picked up either. Is this a bug (in which case I could open an issue) or am I doing something wrong?

**Reply by andrew-codechimp** [reply]
Can you make sure you have updated to the latest version of battery notes (3.2.2). I did see a bug with discovery a while ago that I fixed in this release. 

If that still doesn't fix it can you give me the exact details from the top of the manual add battery entry screen.

**Reply by hrabbach** [reply]
definitely on 3.2.2:

<img width="453" height="176" alt="image" src="https://github.com/user-attachments/assets/2e6fa2fc-e5ff-413c-b0ab-c19c16abb74f" />

This one was auto-detected when I installed Battery Notes:

<img width="501" height="858" alt="image" src="https://github.com/user-attachments/assets/15ac1dbe-2213-4d4f-a20b-85287b571590" />

and adding the same model manually because it wasn't auto-detected:

<img width="497" height="765" alt="image" src="https://github.com/user-attachments/assets/ce996290-5b8b-4990-8069-ba40b4f1b661" />

Same thing for window sensors - an existing one that was auto-detected:

<img width="505" height="864" alt="image" src="https://github.com/user-attachments/assets/290952c9-0627-4d63-b55d-a97a015dfe0d" />

and the same model, adding manually because it's not auto-detected:

<img width="504" height="771" alt="image" src="https://github.com/user-attachments/assets/c99b8abb-b3d0-4c21-8fc2-aaa260a9d86b" />

Thanks for looking into this and the quick response :)

**Reply by hrabbach** [reply]
(just seeing that the model of the window sensor is slightly different - however, I have both generations auto-detected previously)

**Reply by andrew-codechimp** [reply]
Just faked these into my test environment and they discover ok but I have seen something similar before which I thought I'd thoroughly fixed but I will investigate further.
As it's not reproducible easily I don't have a quick fix.
Leave it with me.

**Reply by andrew-codechimp** [reply]
I think I've found the issue with this and 3.2.3 just put into beta should solve it, though it was temperamental so time will tell.
